argparse_dataclass.py

```diff
@@ -359,7 +359,7 @@ def _add_dataclass_options(
         if field.default == field.default_factory == MISSING and not positional:
             kwargs["required"] = True
         else:
-            kwargs["default"] = MISSING
+            kwargs["default"] = field.default
```

Comment:
We need to keep the default as `MISSING` so that the dataclass's initialization handles the default value. This is needed so `default_factory` can work correctly. 

Also, doing it this way makes the initialization the consistent with initializing the dataclass outside of the `argparse_dataclass` package